### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by Yzabakli**
Main classes updated and added new restrictions to username and email address

### CODE REVIEW

File: src/com/company/Customer.java [hunk 1]
```diff
@@ -43,13 +44,25 @@ public Customer(UUID id) {
     }
 
     public void setUserName(String userName) {
-        this.userName = userName;
+
+        if (StaticConstants.CUSTOMER_LIST.stream().map(Customer::getUserName).anyMatch(s -> s.equals(userName))){
+
+            System.err.println("This user name already taken");
+
+            setEmail(new Scanner(System.in).nextLine());
```
Comment (ihccik) [inline src/com/company/Customer.java:52]: why we are calling setEmail method if the user name was already taken?

**Review by ihccik — CHANGES_REQUESTED** [review]
needs fix

**Comment by ihccik** [comment]
branch name should be more meaningful like (DJR3-extension)

File: src/com/company/Customer.java (hunk 1 above)
Comment (Yzabakli) [reply src/com/company/Customer.java]: good catch!

**Pull request closed without merging**